sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/sasl.py

```diff
@@ -111,13 +111,11 @@ class SASLWithWebSocket(WebSocketTransport, SASLTransportMixin):
     def __init__(self, host, credential, port=WEBSOCKET_PORT, connect_timeout=None, ssl=None, **kwargs):
         self.credential = credential
         ssl = ssl or True
-        http_proxy = kwargs.pop('http_proxy', None)
```

Comment:
Good catch !!

Can't really test this because of the reasons you outlined (though there are convos about adding some proxy based tests into the pipeline).

The other reason is that just creating a client simply keeps the proxy information inside `client._config.http_proxy` and that it does properly. Since we lazily create Connections, nothing happens till you do something like `create_batch` etc. Its then when proxy stuff gets passed around and its truly right as its connecting inside `WebSocketTransport` that it looks for whats sent in and tries to apply it.

I can add in a unit test in to pyamqp so that we can make sure the behavior is maintained over changes.

Reply:
that makes sense. one other idea: would a negative test be possible? for ex, if you pass in an invalid http proxy arg, it would raise the correct error since you haven't actually set up a proxy. This would specifically test against the error that you're fixing, which was that it was sending anyway despite http proxy arg being invalid.